igvc_navigation/src/mapper/main.cpp

```diff
@@ -13,221 +15,281 @@
+            if (published_map->at<uchar>(point_x, point_y) <= UCHAR_MAX - (uchar) increment_step) {
+                published_map->at<uchar>(point_x, point_y) += (uchar) increment_step;
+            } else {
+                published_map->at<uchar>(point_x, point_y) = UCHAR_MAX;
+            }
+        } else {
+            offMapCount++;
+        }
     }
-    else
-    {
-      ROS_ERROR_STREAM("\n\nfailed to find transform using empty transform\n\n");
-      tf::StampedTransform transform;
-      transform.setOrigin(tf::Vector3(0.0, 0.0, 0.0));
-      tf::Quaternion q;
-      q.setRPY(0, 0, 0);
-      transform.setRotation(q);
-      transform.child_frame_id_ = "/base_footprint";
-      transform.frame_id_ = "/lidar";
-      transform.stamp_ = ros::Time::now();
-      transforms.insert(std::pair<std::string, tf::StampedTransform>(topic, transform));
+    if (offMapCount > 0) {
+        ROS_WARN_STREAM(offMapCount << " points were off the map");
     }
-  }
-  pcl_ros::transformPointCloud(*msg, *transformed, transforms.at(topic));
+}
 
-  pcl::PointCloud<pcl::PointXYZ>::const_iterator point;
+void checkExistsStaticTransform(const pcl::PointCloud<pcl::PointXYZ>::ConstPtr &msg, const std::string &topic)
+{
+    if (transforms.find(topic) == transforms.end()) {
+        // Wait for transform between frame_id (ex. /scan/pointcloud) and base_footprint.
+        ros::Time messageTimeStamp;
+        pcl_conversions::fromPCL(msg->header.stamp, messageTimeStamp);
+        if (tf_listener->waitForTransform("/base_footprint", msg->header.frame_id, messageTimeStamp, ros::Duration(3.0))) {
+            ROS_INFO_STREAM("\n\ngetting transform for " << topic << "\n\n");
+            tf::StampedTransform transform;
+            tf_listener->lookupTransform("/base_footprint", msg->header.frame_id, messageTimeStamp, transform);
+            transforms.insert(std::pair<std::string, tf::StampedTransform>(topic, transform));
+        } else {
+            ROS_ERROR_STREAM("\n\nfailed to find transform using empty transform\n\n");
+        }
+    }
+}
 
-  for (point = transformed->begin(); point < transformed->points.end(); point++)
-  {
-    double x_loc, y_loc;
-    std::tie(x_loc, y_loc) = rotate(point->x, point->y);
+void decayMap(const ros::TimerEvent&) {
+    int nRows = published_map->rows;
+    int nCols = published_map->cols;
 
-    int point_x = static_cast<int>(std::round(x_loc / resolution + cur_x / resolution + start_x));
-    int point_y = static_cast<int>(std::round(y_loc / resolution + cur_y / resolution + start_y));
-    if (point_x >= 0 && point_y >= 0 && point_x < length_y && start_y < width_x)
+    if (published_map->isContinuous())
     {
-      if (published_map->at<uchar>(point_x, point_y) < 230)
-      {
-        published_map->at<uchar>(point_x, point_y) += (uchar)125;
-      }
-      count++;
+        nCols *= nRows;
+        nRows = 1;
     }
-    else if (!offMap)
+    int i, j;
+    uchar* p;
+    for (i = 0; i < nRows; i++)
```

Comment:
Unrelated to code exactly. But I think we should have discussion on how we should handle the decay.

Reply:
Same thought as @Codetector1374

Reply:
@JasonGibson274 I think we've come to a conclusion on slack that for now decaying globally is fine right?